tst/regression/test_suites/cluster_hydro_agn_feedback/cluster_hydro_agn_feedback.py

```diff
@@ -0,0 +1,470 @@
+                        ],
+                        [1, 0],
+                    )
+                ).v
+
+                drho = inside_jet * agn_kinetic_fraction * time * jet_density
+                dMx = (
+                    inside_jet
+                    * agn_kinetic_fraction
+                    * time
+                    * sign_jet
+                    * jet_density
+                    * jet_velocity
+                    * jet_coords.jet_n[0]
+                )
+                dMy = (
+                    inside_jet
+                    * agn_kinetic_fraction
+                    * time
+                    * sign_jet
+                    * jet_density
+                    * jet_velocity
+                    * jet_coords.jet_n[1]
+                )
+                dMz = (
+                    inside_jet
+                    * agn_kinetic_fraction
+                    * time
+                    * sign_jet
+                    * jet_density
+                    * jet_velocity
+                    * jet_coords.jet_n[2]
+                )
+                dE = (
+                    inside_jet
+                    * agn_kinetic_fraction
+                    * time
+                    * 0.5
+                    * jet_density
+                    * jet_velocity**2
+                )
+
+                return drho, dMx, dMy, dMz, dE
+
+            def thermal_feedback(Z, Y, X, time):
+                if not hasattr(time, "units"):
+                    time = unyt.unyt_quantity(time, "code_time")
+                R = np.sqrt(X**2 + Y**2 + Z**2)
+                inside_sphere = np.piecewise(
+                    R,
+                    [
+                        R <= self.agn_thermal_radius.in_units("code_length"),
+                    ],
+                    [1, 0],
+                )
+                dE = (
+                    inside_sphere
+                    * time
+                    * (
+                        self.fixed_power
+                        * agn_thermal_fraction
+                        / (4.0 / 3.0 * np.pi * self.agn_thermal_radius**3)
+                    )
+                )
+
+                drho = (
+                    inside_sphere
+                    * time
+                    * (
+                        self.fixed_power
+                        / (self.efficiency * unyt.c_cgs**2)
+                        * agn_thermal_fraction
+                        / (4.0 / 3.0 * np.pi * self.agn_thermal_radius**3)
+                    )
+                )
+                # Assume no velocity, no change in momentum with mass injection
+                return drho, dE
+
+            def agn_feedback(Z, Y, X, dt):
+                drho_k, dMx_k, dMy_k, dMz_k, dE_k = kinetic_feedback(Z, Y, X, dt)
+                drho_t, dE_t = thermal_feedback(Z, Y, X, dt)
+
+                drho = drho_k + drho_t
+                dMx = dMx_k
+                dMy = dMx_k
+                dMz = dMx_k
```

Comment:
I got side-tracked while figuring what's wrong with this test. Apparently nothing is injected/changed in either the data read from the AthenaPK output nor the analytical expectation. I don't think the simulation code is broken since I have movies of working kinetic jets, something about the test or comparison is going wrong. I'll have to revisit what exactly is broken in this test later (soon).

Reply:
Fixed this bug and figured out why the test parameters were injecting too weak a jet to test. Everything is working as intended now.